packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -691,31 +691,42 @@ augmentFieldsDict(TagRels, {
 });
 
 defineQuery({
-  name: "TagsByParentId",
+  name: "TagsByCoreTagId",
   schema: `
     type TagWithTotalCount {
       tags: [Tag!]!
       totalCount: Int!
     }
   `,
   resultType: "TagWithTotalCount!",
-  argTypes: "(parentTagId: String, limit: Int, searchTagIds: [String])",
+  argTypes: "(coreTagId: String, limit: Int, searchTagIds: [String])",
   fn: async (
     _root: void,
     args: {
-      parentTagId: string | null;
+      coreTagId: string | null;
       limit?: number;
       searchTagIds?: string[];
     },
     context: ResolverContext
   ) => {
-    const { parentTagId, limit = 20, searchTagIds } = args;
+    const { coreTagId, limit = 20, searchTagIds } = args;
 
-    const { tags, totalCount } = await context.repos.tags.getTagsByParentTagId(
-      parentTagId,
+    const { tags: unsortedTags, totalCount } = await context.repos.tags.getTagsByCoreTagId(
+      coreTagId,
       limit,
       searchTagIds
     );
+    
+    const tags = searchTagIds?.length
+      ? (() => {
+          const searchTagIdOrder = new Map(searchTagIds.map((id, index) => [id, index]));
+          return [...unsortedTags].sort((a, b) => {
+            const indexA = searchTagIdOrder.get(a._id) ?? Infinity;
+            const indexB = searchTagIdOrder.get(b._id) ?? Infinity;
+            return indexA - indexB;
+          });
+        })()
```

Comment:
Um... if I understand what this is meant to be doing correctly, this can just be... 
```
const unsortedTagMap = keyBy(unsortedTags, '_id');
const sortedTags = searchTagIds.map(tagId => unsortedTagMap[tagId]);
return filterNonnull(sortedTags);
```
?

(And it should probably live in a dedicated function, rather than being inlined.)